### PR TITLE
Cleanup to prepare for using mmap pointer in external memory.

### PR DESCRIPTION
**Pull request by trivialfis**
Extracted from https://github.com/dmlc/xgboost/pull/9315 .

- Update SparseDMatrix comment.
- Use a pointer in the bitfield. We will replace the `std::vector<bool>` in `ColumnMatrix` with bitfield.
- Clean up the page source. The timer is removed as it's not accurate once we swap the mmap pointer into the page.

### CODE REVIEW

File: src/common/bitfield.h [hunk 1]
```diff
@@ -86,13 +90,15 @@ struct BitFieldContainer {
 
  public:
   BitFieldContainer() = default;
-  XGBOOST_DEVICE explicit BitFieldContainer(common::Span<value_type> bits) : bits_{bits} {}
-  XGBOOST_DEVICE BitFieldContainer(BitFieldContainer const& other) : bits_{other.bits_} {}
```
Comment (hcho3) [inline src/common/bitfield.h:90]: Is it your intention that we shouldn't allow copy-construction of `BitFieldContainer` on device?

File: src/common/bitfield.h [hunk 2]
```diff
@@ -70,8 +73,9 @@ struct BitFieldContainer {
   };
 
  private:
-  common::Span<value_type> bits_;
-  static_assert(!std::is_signed<VT>::value, "Must use unsiged type as underlying storage.");
+  value_type* bits_{nullptr};
```
Comment (hcho3) [inline src/common/bitfield.h:76]: Can you explain why we should use a raw pointer here? You mentioned the use of `std::vector<bool>` in `ColumnMatrix`. Does the span behave badly when `value_type == bool` ?

Comment (trivialfis) [reply src/common/bitfield.h:76]: It's used in hot loops, the check used in span can be expensive.

File: src/common/bitfield.h (hunk 1 above)
Comment (trivialfis) [reply src/common/bitfield.h:90]: When `default` is specified, nvcc will treat it as host-device function automatically.

**Review by hcho3 — APPROVED** [review]

**Pull request merged**